File: find bottom left tree values.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        int res=0;
        Queue<TreeNode>q=new LinkedList<TreeNode>();
        q.add(root);
        while(q.size()>0){
            int sz=q.size();
            TreeNode first=null;
            while(sz-->0){
                TreeNode cur=q.peek();q.remove(q.peek());
                if(first==null){first=cur;}
                if(cur.left!=null){q.add(cur.left);}
                if(cur.right!=null){q.add(cur.right);}
            }
            if(first!=null){res=first.val;}
        }
        return res;
    }
}

File: find bottom left tree values.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    int findBottomLeftValue(TreeNode* root) {
       queue<TreeNode*>q;
       q.push(root);
       int res=0;
       while(q.size()){
           int sz=q.size();
           TreeNode* first=nullptr;
           while(sz--){
               TreeNode* cur=q.front();q.pop();
                if(first==nullptr){first=cur;}
                if(cur->left){q.push(cur->left);}
                if(cur->right){q.push(cur->right);}
           }
           if(first!=nullptr){res=first->val;}
       }
       return res;
    }
};